### ISSUE TITLE
Allow editing the message of a stash

### ISSUE THREAD

**Issue by FichteFoll** [sublimehq/sublime_merge#1768]
**Problem description**

It is currently not possible to edit the message of a stash (commit).
SM allows entering edit mode for a stash commit but when trying to submit the changes, it complains about the commit not being reachable from HEAD.

**Preferred solution**

Trying to edit the message of any stash functions exactly like editing a commit's message that is "reachable from HEAD".

**Screenshots**

![2023-06-17_18-38-16](https://github.com/sublimehq/sublime_merge/assets/931051/7de264c9-67ff-42a6-8280-1e8ee70c0748)

**Comment by mobile-ar** [comment]
Yes to this!!! Or even better will be nice to be able to stash with the flag/option of `[(-m | --message) <message>]` to be able to save the stash with whatever message we want to identify the stash.

<img width="1009" alt="image" src="https://github.com/sublimehq/sublime_merge/assets/758096/d9cb94ac-d5a4-45d6-b74b-077d44fd9bd9">
